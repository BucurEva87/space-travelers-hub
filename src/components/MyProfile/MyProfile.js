import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { unbookRocket } from '../../redux/rockets/rockets';
import { unbookDragon } from '../../redux/dragons/dragons';
import { leaveMission } from '../../redux/missions/missions';
import style from './MyProfile.module.scss';

const MyProfile = () => {
  const rockets = useSelector((state) => state.rockets.list).filter((rocket) => rocket.reserved);
  const missions = useSelector((state) => state.missions.list);
  const reservedMissions = missions.filter((mission) => mission.reserved);
  const dragons = useSelector((state) => state.dragons.list).filter((dragon) => dragon.reserved);
  const dispatch = useDispatch();

  return (
    <section className={style.profile} data-testid="myprofile">
      <div className={style.category}>
        <h2>My Missions</h2>
        <ul>
          { reservedMissions.length ? reservedMissions.map((mission) => (
            <li key={mission.id}>
              <span>{mission.name}</span>
              <button
                type="button"
                className={style['btn-unreserve']}
                onClick={() => dispatch(leaveMission(mission.id))}
              >
                Cancel Reservation
              </button>
              <a href={mission.wikipedia} className={style['btn-more']}>More</a>
            </li>
          )) : <li>There is no booked mission</li> }
        </ul>
      </div>
      <div className={style.category}>
        <h2>My Rockets</h2>
        <ul>
          { rockets.length ? rockets.map((rocket) => (
            <li key={rocket.id} data-id={rocket.id}>
              <span>{rocket.name}</span>
              <button
                type="button"
                className={style['btn-unreserve']}
                onClick={(e) => dispatch(unbookRocket(+e.target.closest('li').dataset.id))}
              >
                Cancel Reservation
              </button>
              <a href={rocket.wikipedia} className={style['btn-more']}>More</a>
            </li>
          )) : <li>There is no booked rocket</li> }
        </ul>
      </div>
      <div className={style.category}>
        <h2>My Dragons</h2>
        <ul>
          {dragons.length ? dragons.map((dragon) => (
            <li key={dragon.id} id={dragon.id}>
              <span>{ dragon.name }</span>
              <button
                type="button"
                className={style['btn-unreserve']}
                onClick={() => dispatch(unbookDragon(dragon.id))}
              >
                Cancel Reservation
              </button>
              <a href={dragon.wikipedia} className={style['btn-more']}>More</a>
            </li>
          )) : <li>There is no booked dragon</li> }
        </ul>
      </div>
    </section>
  );
};

export default MyProfile;
