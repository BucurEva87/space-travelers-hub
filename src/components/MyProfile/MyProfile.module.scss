@import '../../styles/variables';

.profile {
  display: flex;
  justify-content: stretch;
  font-size: 1rem;
  gap: 2rem;
}

.category {
  width: 33%;
  display: flex;
  flex-direction: column;
  gap: 2rem;
}

h2 {
  font-size: 1.4rem;
}

ul > li {
  width: 100%;
  display: flex;
  align-items: center;
  margin-bottom: 0.8rem;
}

ul span {
  font-size: 1rem;
  font-weight: bold;
}

.btn-unreserve {
  background-color: $white;
  padding: 0.4rem 1.8rem;
  border: 0.1rem solid $gray;
  color: $gray;
  margin-left: auto;
  border-radius: 0.4rem;
  cursor: pointer;
}

.btn-more {
  padding: 0.2rem 0.4rem;
  background-color: $primaryBtnBlue;
  color: $white;
  border-radius: 0.4rem;
  font-size: 1rem;
  margin-left: 1rem;
  cursor: pointer;
}
