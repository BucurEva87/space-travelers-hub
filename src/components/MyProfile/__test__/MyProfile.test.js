import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import MyProfile from '../MyProfile';
import store from '../../../redux/configureStore';

describe('MyProfile component', () => {
  test('renders on the screen', () => {
    expect(render(
      <Provider store={store}>
        <MyProfile />
      </Provider>,
    ).getByTestId('myprofile')).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <MyProfile />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
