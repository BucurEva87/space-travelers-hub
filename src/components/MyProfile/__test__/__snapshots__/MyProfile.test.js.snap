// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfile component matches snapshot 1`] = `
<section
  className="profile"
  data-testid="myprofile"
>
  <div
    className="category"
  >
    <h2>
      My Missions
    </h2>
    <ul />
  </div>
  <div
    className="category"
  >
    <h2>
      My Rockets
    </h2>
    <ul>
      <li>
        There is no booked rocket
      </li>
    </ul>
  </div>
  <div
    className="category"
  >
    <h2>
      My Dragons
    </h2>
    <ul>
      <li>
        There is no booked dragon
      </li>
    </ul>
  </div>
</section>
`;
